src/libraries/Common/src/Interop/Windows/Kernel32/Interop.CreateFileMapping.cs

```diff
@@ -5,12 +5,41 @@
 #nullable enable
 using Microsoft.Win32.SafeHandles;
 using System;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 
 internal partial class Interop
 {
     internal partial class Kernel32
     {
+        public static SafeMemoryMappedFileHandle CreateFileMapping(
+                SafeFileHandle hFile,
+                ref SECURITY_ATTRIBUTES securityAttributes,
+                int pageProtection,
+                long maximumSize,
+                string? name)
+        {
+            // split the long into two ints
+            int capacityHigh, capacityLow;
+            SplitLong(maximumSize, out capacityHigh, out capacityLow);
+
+            return CreateFileMapping(hFile, ref securityAttributes, pageProtection, capacityHigh, capacityLow, name);
+        }
+
+        public static SafeMemoryMappedFileHandle CreateFileMapping(
+                IntPtr hFile,
+                ref SECURITY_ATTRIBUTES securityAttributes,
+                int pageProtection,
+                long maximumSize,
+                string? name)
+        {
+            // split the long into two ints
+            int capacityHigh, capacityLow;
+            SplitLong(maximumSize, out capacityHigh, out capacityLow);
+
+            return CreateFileMapping(hFile, ref securityAttributes, pageProtection, capacityHigh, capacityLow, name);
```

Comment:
```suggestion
            return CreateFileMapping(hFile, ref securityAttributes, pageProtection, dwMaximumSizeHigh: (int)(maximumSize >> 32), dwMaximumSizeLow: (int)maximumSize , name);
```

Now that the split is in the wrapper, I think it is better to do it inline. The extra ceremony from the SplitLong method does seems worth it. It is just a pile of busy work for the JIT to optimize out.